src/components/home/Courses.js

```diff
@@ -0,0 +1,165 @@
+          {sessionStorage.getItem('TOKEN') ? (
+            'Please select a course category'
+          ) : (
+            <span className="text-rose-500 font-bold">
+              Please login to continue...
+            </span>
+          )}
+        </h6>
+
+        <div className="flex items-center justify-between">
+          <div className="border rounded-r-full px-7 py-3 border-gray-300 bg-gray-300 hover:border-lime-500 hover:bg-lime-500">
+            <BsCaretLeft
+              onClick={slideRight}
+              style={{ color: 'white', marginRight: '-1rem' }}
+              size={30}
+            />
+          </div>
+
+          <div className="overflow-x-scroll scroll whitespace-nowrap scroll-smooth scrollbar-hide">
+            <div
+              id="slider"
+              className="flex items-center overflow-x-scroll scroll whitespace-nowrap scroll-smooth scrollbar-hide"
+            >
+              {state.length > 0
+                ? state.map((item) => (
+                  <div
+                    key={item.id}
+                    className="mx-20 cursor-pointer hover:scale-105 ease-in-out duration-300"
+                  >
+                    {imageDisplay(item)}
+
+                    <p className="mt-5 block text-center">
+                      <span className="text-xl font-extrabold">
+                        {item.course_type}
+                        {' '}
+                        :
+                      </span>
+                      <br />
+                      <br />
+                      <span className="text-sm text-gray-300">
+                        *******************************
+                      </span>
+                      <br />
+                      <span className="text-sm text-gray-600">
+                        {item.info}
+                      </span>
+                    </p>
+                    <div className="mt-5 flex text-gray-400 justify-center">
+                      <TiSocialFacebook
+                        size={40}
+                        className="mr-5 py-2 rounded-full border border-gray-300"
+                      />
+                      <TiSocialTwitter
+                        size={40}
+                        className="mr-5 py-2 rounded-full border border-gray-300"
+                      />
+                      <TiSocialInstagram
+                        size={40}
+                        className="mr-5 py-2 rounded-full border border-gray-300"
+                      />
+                    </div>
+                  </div>
+                ))
+                : null}
+
+              {state.length === 0 ? (
+                <div>
+                  <img className="w-full" src={plane} alt="/" />
+                  <p className="font-thin text-center mb-20">
+                    No course added, yet.
+                  </p>
+                </div>
+              ) : null}
+              {sessionStorage.getItem('TOKEN') ? null : (
+                <div>
```

Comment:
There are some candidates to make some of these html components reusable.

Reply:
Thank you. Refactored both desktop & mobile components.